pb/edge_cmd_pb/impl.go

```diff
@@ -83,78 +87,159 @@ func DecodeTags(tags map[string]*TagValue) map[string]interface{} {
 	return result
 }
 
-func EncodeJson(tags map[string]interface{}) (*JsonMap, error) {
+func EncodeJsonMap(tags map[string]interface{}) (*JsonMap, error) {
 	if len(tags) == 0 {
 		return nil, nil
 	}
 	result := map[string]*JsonValue{}
 
 	for k, v := range tags {
-		if v == nil {
-			result[k] = &JsonValue{
-				Value: &JsonValue_NilValue{
-					NilValue: true,
-				},
-			}
-		} else {
-			switch val := v.(type) {
-			case string:
-				result[k] = &JsonValue{
-					Value: &JsonValue_StringValue{
-						StringValue: val,
-					},
-				}
-			case bool:
-				result[k] = &JsonValue{
-					Value: &JsonValue_BoolValue{
-						BoolValue: val,
-					},
-				}
-			case float64:
-				result[k] = &JsonValue{
-					Value: &JsonValue_FpValue{
-						FpValue: val,
-					},
-				}
-			case map[string]interface{}:
-				mapVal, err := EncodeJson(val)
-				if err != nil {
-					return nil, err
-				}
-				result[k] = &JsonValue{
-					Value: &JsonValue_MapValue{
-						MapValue: mapVal,
+		encoded, err := EncodeJsonValue(v)
+		if err != nil {
+			return nil, err
+		}
+		result[k] = encoded
+	}
+	return &JsonMap{Value: result}, nil
+}
+
+func EncodeJsonValue(v interface{}) (*JsonValue, error) {
```

Comment:
I don't like this. I don't understand why we need this.

Reply:
It's not used any more and I was going to remove it in a follow up, but I can remove it now

Reply:
removed